test/Microsoft.DotNet.PackageValidation.Tests/Filtering/TargetFrameworkFilterTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.PackageValidation.Filtering;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.PackageValidation.Tests.Filtering
+{
+    public class TargetFrameworkFilterTests
```

Comment:
Please add a test for passing "*" as suggested by Eric: https://github.com/dotnet/sdk/pull/38374/files#r1473573127